packages/backend/src/peripherals/database/migrations/037_preprocessed_l2_transaction.ts

```diff
@@ -0,0 +1,47 @@
+/*
+                      ====== IMPORTANT NOTICE ======
+
+DO NOT EDIT OR RENAME THIS FILE
+
+This is a migration file. Once created the file should not be renamed or edited,
+because migrations are only run once on the production server. 
+
+If you find that something was incorrectly set up in the `up` function you
+should create a new migration file that fixes the issue.
+
+*/
+import { Knex } from 'knex'
+
+export async function up(knex: Knex) {
+  await knex.schema.alterTable('preprocessed_state_details', (table) => {
+    table.jsonb('l2_transactions_statistics')
+    table.jsonb('cumulative_l2_transactions_statistics')
+  })
+
+  await knex.schema.createTable(
+    'preprocessed_user_l2_transactions_statistics',
+    (table) => {
+      table.increments('id').primary()
+      table
+        .integer('state_update_id')
+        .notNullable()
+        .references('state_update_id')
+        .inTable('preprocessed_state_updates')
+      table.string('stark_key').notNullable()
+      table.jsonb('l2_transactions_statistics').notNullable()
+      table.jsonb('cumulative_l2_transactions_statistics').notNullable()
+      table.index(['stark_key', 'state_update_id'])
```

Comment:
The order of columns in the index is important. Since we often select by `state_update_id` only, it should come first, because if it comes second, this index won't be used (unless something changed since I learned about pg indexes :)).

BTW, we still need to spend some time afterwards and make sure we're not missing any indexes, and the ones we have are utilized and don't have too much data.

Reply:
What if we query by stark key as often as state update id? 😅